components/gsma-authenticators/src/main/java/com/wso2telco/gsma/authenticators/voice/VoiceCallAuthenticator.java

```diff
@@ -0,0 +1,265 @@
+
+        String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(context.getQueryParams(),
+                context.getCallerSessionKey(), context.getContextIdentifier());
+        ApplicationConfig applicationConfig = context.getSequenceConfig().getApplicationConfig();
+        Map<String, String> paramMap = Util.createQueryParamMap(queryParams);
+        String msisdn = (String) context.getProperty(MSISDN);
+        String clientId = paramMap.get(CLIENT_ID);
+        String applicationName = applicationConfig.getApplicationName();
+        String sessionKey = context.getCallerSessionKey();
+
+        log.info("MSISDN : " + msisdn);
+        log.info("Client ID : " + clientId);
+        log.info("Application name : " + applicationName);
+        log.info("sessionKey  : " + sessionKey);
+
+        ValidSoftJsonHelper validSoftJsonHelper = new ValidSoftJsonHelper();
+        validSoftJsonHelper.setIsUserActiveRequestJson(sessionKey, msisdn);
+        validSoftJsonHelper.setUserRegistrationAndAuthenticationJson(sessionKey, msisdn);
+        StringEntity postData = null;
+        StringEntity verifyAndRegistartionPostData = null;
+        try {
+            postData = new StringEntity(validSoftJsonHelper.getIsUserActiveRequestJson());
+            verifyAndRegistartionPostData = new StringEntity(validSoftJsonHelper.getUserRegistrationAndAuthenticationJson());
+            log.info("Json Object : " + postData);
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+
+        String url = "/authenticationendpoint/mcx-user-registration/ivr_waiting" + "?" + queryParams + "&redirect_uri=" +
+                (String) context.getProperty("redirectURI") + "&authenticators="
+                + getName() + ":" + "LOCAL" + ""+"&sessionDataKey=" + sessionKey;
+
+        try {
+            boolean isUserEnrolledInValidSoft = verifyIsUserActiveFromValidSoftServer(isUserEnrolledUrl, postData);
+            context.setProperty(IS_FLOW_COMPLETED, false);
+            DBUtils.insertAuthFlowStatus(msisdn, Constants.STATUS_PENDING, context.getContextIdentifier());
+            if (isUserEnrolledInValidSoft) {
+                VoiceIVRFutureCallback futureCallback = new VoiceIVRFutureCallback(context,msisdn);
+                postRequest(verifyUserUrl, verifyAndRegistartionPostData ,
+                        futureCallback);
+                response.sendRedirect(url);
+            } else {
+                VoiceIVRFutureCallback futureCallback = new VoiceIVRFutureCallback(context,msisdn);
+                postRequest(onBoardUserUrl, verifyAndRegistartionPostData ,
+                        futureCallback);
+                response.sendRedirect(url);
+            }
+        } catch (IOException e) {
+            log.error("Error occurred while redirecting request", e);
+        } catch(JSONException je){
+            je.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS1148)